plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseCastPushdownTest.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.Session;
+import io.trino.sql.planner.plan.ProjectNode;
+import io.trino.sql.query.QueryAssertions;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.sql.SqlExecutor;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.util.List;
+
+import static io.trino.plugin.jdbc.CastDataTypeTest.CastTestCase;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+@TestInstance(PER_CLASS)
+public abstract class BaseCastPushdownTest
```

Comment:
Can we follow the testing guidelines as mentioned here - https://trino.io/docs/current/develop/tests.html

Reply:
Can it extend`AbstractTestQueryFramework` ? Since we it provides a lot of helper/util methods for assertion -

Reply:
If not it could be like a `SqlDataTypeTest`

Reply:
As of now used `AbstractTestQueryFramework` till we decide to use `Nested`.

CC: @ssheikin